packages/fxa-auth-server/test/local/routes/account.js

```diff
@@ -2884,99 +2893,84 @@ describe('/account/destroy', () => {
+      return [{ status: 'Active', billingAgreementId: 'B-test' }];
+    });
+    mockAuthModels.deleteAllPayPalBAs = sinon.spy(async (uid) => {
+      return;
+    });
   });
 
   function buildRoute(subscriptionsEnabled = true) {
-    const accountRoutes = makeRoutes({
-      checkPassword: function () {
-        return Promise.resolve(true);
-      },
-      config: {
-        subscriptions: {
-          enabled: subscriptionsEnabled,
+    Container.set(StripeHelper, mockStripeHelper);
+    Container.set(PayPalHelper, mockPaypalHelper);
+    const accountRoutes = makeRoutes(
+      {
+        checkPassword: function () {
+          return Promise.resolve(true);
+        },
+        config: {
+          subscriptions: {
+            enabled: subscriptionsEnabled,
+            paypalNvpSigCredentials: {
+              enabled: true,
+            },
+          },
+          domain: 'wibble',
         },
-        domain: 'wibble',
+        db: mockDB,
+        log: mockLog,
+        push: mockPush,
+        stripeHelper: mockStripeHelper,
       },
-      db: mockDB,
-      log: mockLog,
-      push: mockPush,
-      stripeHelper: mockStripeHelper,
-    });
+      { 'fxa-shared/db/models/auth': mockAuthModels }
+    );
     return getRoute(accountRoutes, '/account/destroy');
   }
 
   it('should delete the account', () => {
     const route = buildRoute();
 
     return runTest(route, mockRequest, () => {
-      assert.equal(
-        mockDB.accountRecord.callCount,
-        1,
-        'db.emailRecord was called once'
-      );
-      let args = mockDB.accountRecord.args[0];
-      assert.equal(args.length, 1);
-      assert.equal(args[0], email);
-
-      assert.equal(
-        mockDB.deleteAccount.callCount,
-        1,
-        'db.deleteAccount was called once'
-      );
-      args = mockDB.deleteAccount.args[0];
-      assert.equal(args.length, 1, 'db.deleteAccount was passed one argument');
-      assert.equal(
-        args[0].email,
-        email,
-        'db.deleteAccount was passed email record'
-      );
-      assert.deepEqual(args[0].uid, uid, 'email record had correct uid');
-
-      assert.equal(mockStripeHelper.removeCustomer.callCount, 1);
-      args = mockStripeHelper.removeCustomer.args[0];
-      assert.lengthOf(args, 2);
-      assert.equal(args[0], uid);
-
-      assert.equal(mockPush.notifyAccountDestroyed.callCount, 1);
-      assert.equal(mockPush.notifyAccountDestroyed.firstCall.args[0], uid);
-
-      assert.equal(
-        mockLog.notifyAttachedServices.callCount,
-        1,
-        'log.notifyAttachedServices was called once'
-      );
-      args = mockLog.notifyAttachedServices.args[0];
-      assert.equal(
-        args.length,
-        3,
-        'log.notifyAttachedServices was passed three arguments'
-      );
-      assert.equal(args[0], 'delete', 'first argument was event name');
-      assert.equal(args[1], mockRequest, 'second argument was request object');
-      assert.equal(
-        args[2].uid,
```

Comment:
I so enjoy trimming this legacy test stuff for the nicer, compact, and equivilant `sinon.assert` statements.